components/tag/stories/template.js

```diff
@@ -35,32 +40,50 @@ export const Template = ({
 					typeof size !== "undefined",
 				"is-emphasized": isEmphasized,
 				"is-disabled": isDisabled,
-				"is-invalid": isInvalid,
 				"is-selected": isSelected,
+				"is-hover": isHovered,
+				"is-focus-visible": isFocused,
+				"is-active": isActive,
+				"is-readOnly": isReadOnly,
 				...customClasses.reduce((a, c) => ({ ...a, [c]: true }), {}),
 			})}
 			id=${ifDefined(id)}
 			tabindex=${isDisabled ? "-1" : "0"}
 			style=${styleMap(customStyles)}
+			@click=${function() {
+				if (isReadOnly || isDisabled) return;
+				updateArgs({ isSelected: !isSelected });
+			}}
+			@focusin=${function() {
+				if (isReadOnly || isDisabled) return;
+				updateArgs({ isFocused: true });
+			}}
+			@focusout=${function() {
+				updateArgs({ isFocused: false });
+			}}
 		>
-			${when(avatarUrl && !isInvalid, () =>
+			${when(avatarUrl, () =>
 				Avatar({
 					image: avatarUrl,
-					size: "50",
 				}, context)
 			)}
-			${when(iconName || isInvalid, () =>
+			${when(iconName, () =>
 				Icon({
 					size,
 					iconName,
 					setName: "workflow",
 					customClasses: [`${rootClass}-itemIcon`],
 				}, context)
 			)}
+			${when(thumbnailUrl, () =>
+				Thumbnail({
+					imageURL: thumbnailUrl,
+				}, context)
+			)}
 			<span class="${rootClass}-itemLabel">${label}</span>
 			${when(hasClearButton, () =>
 				ClearButton({
-					size,
```

Comment:
Many of these (clear button, avatar, thumbnail) have their size set by CSS, there doesn't seem to be a lot of value to passing down the size in the template, but let me know if it seems like I'm missing something.

Reply:
I think this question about which approach is better was the origin of why we decided that it would be nice to have a style guide (and it was also in relation to this tag migration the first time around). I'll take a look at this too, I think my initial thought is always "but we have a token for it, so we should use it" and it helps ensure that we get it shipped with the right size, but on the other hand, extra unnecessary code isn't always the best thing. I'll take another look at this one too.

Reply:
Ok, it is size-mapped in the template... but we're also using the tokens. Overkill? It feels like putting in the CSS is nice, because the visuals will be appropriately sized without needing to add it in the implementation, but is there an argument against that?

Reply:
I agree! We should probably keep it in the CSS to ensure the sizes get defined with the tokens. 👍